### PR TITLE
Blog: Helm integration with odo

### LINKED ISSUE #6645: Blog Post: pick a use case for using exec for Deploy (e.g., DB Migration, Helm, Kustomize, ..)

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**
/area documentation

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6645 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [x] Documentation 

**How to test changes / Special notes to the reviewer:**
https://deploy-preview-6775--odo-docusaurus-preview.netlify.app/blog/using-helm-with-odo

### CODE REVIEW

**Review by rm3l — CHANGES_REQUESTED** [review]
```
1:27:19 PM: [ERROR] Unable to build website for locale en.
1:27:19 PM: [ERROR] Error: Docusaurus found broken links!
1:27:19 PM: 
1:27:19 PM: Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.
1:27:19 PM: Exhaustive list of all broken links found:
1:27:19 PM: 
1:27:19 PM: - On source page path = /blog:
1:27:19 PM:    -> linking to github.com/redhat-developer/odo/releases/tag/v3.8.0 (resolved as: /github.com/redhat-developer/odo/releases/tag/v3.8.0)
1:27:19 PM: 
1:27:19 PM: - On source page path = /blog/using-helm-with-odo:
1:27:19 PM:    -> linking to github.com/redhat-developer/odo/releases/tag/v3.8.0 (resolved as: /blog/github.com/redhat-developer/odo/releases/tag/v3.8.0)
1:27:19 PM: 
1:27:19 PM:     at throwError (/opt/build/repo/docs/website/node_modules/@docusaurus/logger/lib/index.js:76:11)
1:27:19 PM:     at handleBrokenLinks (/opt/build/repo/docs/website/node_modules/@docusaurus/core/lib/server/brokenLinks.js:153:47)
1:27:19 PM:     at async buildLocale (/opt/build/repo/docs/website/node_modules/@docusaurus/core/lib/commands/build.js:186:5)
1:27:19 PM:     at async tryToBuildLocale (/opt/build/repo/docs/website/node_modules/@docusaurus/core/lib/commands/build.js:41:20)
1:27:19 PM:     at async mapAsyncSequential (/opt/build/repo/docs/website/node_modules/@docusaurus/utils/lib/jsUtils.js:34:24)
1:27:19 PM:     at async Command.build (/opt/build/repo/docs/website/node_modules/@docusaurus/core/lib/commands/build.js:76:21)
```

Looks like there are broken links in this blog post..

**Comment by feloy** [comment]
Could you add a `<!--truncate-->` line somewhere, or the complete article is displayed in the blogs page?

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md [hunk 1]
```diff
@@ -0,0 +1,815 @@
...
+          matchLabels:
+            app: {{RESOURCE_NAME}}
+        template:
+          metadata:
+            labels:
+              app: {{RESOURCE_NAME}}
+          spec:
+            containers:
+              - name: {{RESOURCE_NAME}}
+ image: {{CONTAINER_IMAGE}}
```
Comment (rm3l) [inline docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:471]: Looks like this (and all the container spec definition below) is not indented correctly..

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md [hunk 2]
```diff
@@ -0,0 +1,815 @@
...
```
Comment (rm3l) [inline docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:228]: Same point: indentation issues..

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md [hunk 3]
```diff
@@ -0,0 +1,815 @@
...
+      apiVersion: networking.k8s.io/v1
+      kind: Ingress
+      metadata:
+        name: {{RESOURCE_NAME}}
+      spec:
+        rules:
+          - host: "{{DOMAIN_NAME}}"
+            http:
+              paths:
+ - path: "/"
```
Comment (rm3l) [inline docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:521]: Same point: indentation issues..

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md [hunk 4]
```diff
@@ -0,0 +1,815 @@
...
+title: Using Helm with odo
+author: Parthvi Vala
+author_url: https://github.com/valaparthvi
+author_image_url: https://github.com/valaparthvi.png
+tags: []
+slug: using-helm-with-odo
+---
+
+This blog will show how odo can now be used with tools such as Helm, Kustomize, etc. for the [outerloop](/docs/introduction/#what-is-inner-loop-and-outer-loop) development cycle.
+
```
Comment (valaparthvi) [inline docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:11]:
```suggestion
<!--truncate-->
```

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md [hunk 5]
```diff
@@ -0,0 +1,815 @@
...
+  CONTAINER_IMAGE: quay.io/MYUSERNAME/go-odo-example
+  RESOURCE_NAME: my-go-app
+  CONTAINER_PORT: "8080"
+  DOMAIN_NAME: go.example.com
+  USERNAME: root
+  HOST: mongodb
+```
+
+:::note
+Ensure that you replace `MYUSERNAME` in `CONTAINER_IMAGE` with your own username; or use a container registry that you have the write permissions to.
```
Comment (rm3l) [inline docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:303]: Without changing the Devfile, can you leverage variable substitution instead? By running `odo deploy --var CONTAINER_IMAGE=...`, I think it makes things easy to manage, even from the user perspective, if they want to push to a completely different registry, not just `quay.io`..

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md [hunk 6]
```diff
@@ -0,0 +1,815 @@
...
+We will first define the `deploy-db` command that is used to deploy the helm chart.
+
+To use an external tool such as helm or kustomize, we need to ensure 2 things:
+1. use an `exec` command; learn more [here](/docs/development/devfile#how-odo-runs-exec-commandsin-deploy-mode).
+2. the container component referenced by this command uses an image that contains the required binary.
+
+```yaml
+- id: deploy-db
+  exec:
+    commandLine: helm repo add bitnami https://charts.bitnami.com/bitnami && helm repo update && helm install mongodb bitnami/mongodb
```
Comment (rm3l) [inline docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:111]: Suggestion to run `helm upgrade --install` instead, so it does not fail if we need to run it again.

```suggestion
    commandLine: helm repo add bitnami https://charts.bitnami.com/bitnami && helm repo update && helm upgrade --install mongodb bitnami/mongodb
```

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md [hunk 7]
```diff
@@ -0,0 +1,815 @@
...
+    image: quay.io/tkral/devbox-demo-devbox
+  attributes:
+    pod-overrides:
+      spec:
+        serviceAccountName: my-go-app
+```
+
+The image used by this container component contains the Helm binary that we can use to deploy the helm chart.
+
+The component is using a `pod-overrides` attribute that will override the service account used by the pod to deploy the helm chart to use the service account (`my-go-app`) we define in this Devfile.
```
Comment (rm3l) [inline docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:158]: Can you link to the `pod-overrides` documentation to learn more: https://devfile.io/docs/2.2.0/overriding-pod-and-container-attributes#pod-overrides

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md [hunk 8]
```diff
@@ -0,0 +1,815 @@
...
+We will first add the component referenced by `deploy-db` command.
+
+```yaml
+- name: deploy-db
+  container:
+    image: quay.io/tkral/devbox-demo-devbox
+  attributes:
+    pod-overrides:
+      spec:
+        serviceAccountName: my-go-app
```
Comment (rm3l) [inline docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:153]: I guess #6764 prevents us from using the same `RESOURCE_NAME` variable here, right?

**Review by rm3l — CHANGES_REQUESTED** [review]

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md (hunk 6 above)
Comment (valaparthvi) [reply docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:111]: :100:
I kept running into failure when mongodb was already installed.

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md (hunk 8 above)
Comment (valaparthvi) [reply docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:153]: Yes. This blog was how I discovered that issue actually.

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md (hunk 2 above)
Comment (valaparthvi) [reply docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:228]: I am not sure when this got messed up. I remember copy pasting the commands as they are and testing them, and it worked fine for me then.

File: docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md (hunk 5 above)
Comment (valaparthvi) [reply docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md:303]: Oh, you're right, that is a good idea.

**Review by rm3l — APPROVED** [review]

**Comment by rm3l** [comment]
/override ODO-PODMAN-TEST
/override Kubernetes-Integration-Tests/Kubernetes-Integration-Tests
/override windows-integration-test/Windows-test

Not related - doc-only.

**Pull request merged**